life_expectancy/tests/test_load_save_data.py

```diff
@@ -0,0 +1,22 @@
+"""Test for load_data function"""
+from life_expectancy.clean_data.utils.load_save_data import (
+    load_data, save_data
+)
+from . import FIXTURES_DIR
+import pandas as pd
+from unittest.mock import patch
+
+
+def test_load_data():
+    """Test the data loading"""
+    df_input = load_data(FIXTURES_DIR / "eu_life_expectancy_raw.tsv")
+    assert df_input.shape == (100, 63)
+
+
+@patch("pandas.DataFrame.to_csv")
+def test_save_data(patched_to_csv):
```

Comment:
I think you forgot to do the third point: "Then, assert that the pd.DataFrame.to_csv method is being called. As a bonus, by setting up your tests in this fashion, we can ensure no data transformations modifies any actual data."

With the help of ChatGPT, I have done it in this way:
```
# expected output_file name/directory
output_file = str(Path(OUTPUT_DIR, 'testregion_life_expectancy.csv'))

# Assert that the function was called with the correct arguments
patched_DataFrame_to_csv.assert_called_once_with(output_file, index=False)
```

Looking to your code, I also figured out that I forgot to do the second point of the instructions :).

Reply:
Oh, I thought that the `assert a == "Hello"` was enough (noticed while writing that I forgot to give a meaningful name to the variable 😅).

Because if I'm patching the `pandas.DataFrame.to_csv` to return the value `Hello` I can assess if the method is being called by doing `assert a == "Hello"`, where `a` is the output value of a call to the `save_data` function.

I mean, your solution is much more elegant and I'm going to change mine! But do you think my solution makes sense (although not so obvious as yours)?

Reply:
Applied the fix in [62b3669](https://github.com/2010b9/learning-pod-foundations-assignments/pull/6/commits/62b36697d2e376ea1ca9bfb97007821172b275d4). Indeed your approach is way cleaner than mine – thanks! 🙂